compiler/kernel/proc.go

```diff
@@ -557,3 +570,19 @@ func (b *Builder) LoadConsts(ops []dag.Op) error {
 	}
 	return nil
 }
+
+func evalAtCompileTime(zctx *zson.Context, scope *Scope, in dag.Expr) (zng.Value, error) {
```

Comment:
I think I'll leave it because it's not a generic eval.  When we add generalized constant folding, it will be subsumed by that work and we can change it then.